app/controllers/ncr/proposals_controller.rb

```diff
@@ -29,6 +32,46 @@ def create
       end
     end
 
+    def not_approved
+      cart = Cart.find(params[:id])
+      if cart.approved?
+        redirect_to new_ncr_proposal_path, :alert => "That proposal's already approved. New proposal?"
+      end
+    end
+    def cart_owner
+      cart = Cart.find(params[:id])
+      if cart.requester != current_user
+        redirect_to new_ncr_proposal_path, :alert => 'You cannot restart that proposal'
+      end
+    end
+
+    def edit
+      cart = Cart.find(params[:id])
+      @proposal_form = Ncr::ProposalForm.from_cart(cart)
+      @form_url, @form_method = {action: "update"}, "put"
+      render 'new'
```

Comment:
Can we rename the view to `form.html.erb`? This is the Rails-y way of indicating that it's used for both `new` and `edit`.